DOC-614 | Added `readOwnWrites` option for UPSERT operations (#409)
arangodb/docs-hugo: site/content/3.12/aql/high-level-operations/upsert.md

  OPTIONS { indexHint: … , forceIndexHint: true }
```

### `readOwnWrites`

The `readOwnWrites` option allows `UPSERT` operations to process its inputs one
by one. The default value is `true`. When enabled, the `UPSERT` operation can
observe its own writes and can handle modifying the same target document multiple
times in the same query.

When the option is set to `false`, an `UPSERT` operation processes its inputs

> in batches. A batch has normally 1000 inputs, so it can lead to a faster execution.

Reviewer: ```suggestion
in batches. Normally, a batch has 1000 inputs, which can lead to a faster execution.
```

> However, when using batches, the `UPSERT` operation cannot observe its own writes.

Author: How would you rephrase it?

> However, when using batches, the `UPSERT` operation cannot observe its own writes.

Reviewer: However, when using batches, the `UPSERT` operation can essentially not observe its own writes.

> However, when using batches, the `UPSERT` operation cannot observe its own writes.

Reviewer: ```suggestion
However, when using batches, the `UPSERT` operation can essentially not observe its own writes.
```

> The `readOwnWrites` option should be set to `false` only if it can
> be guaranteed that the input of the `UPSERT` leads to disjoint documents being

Reviewer: ```suggestion
You should only set the `readOwnWrites` option to `false` if you can
guarantee that the input of the `UPSERT` leads to disjoint documents being
```
